src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseService.java

```diff
@@ -113,6 +126,46 @@ public void notifyChangedTestCases(ProgrammingExercise programmingExercise, Obje
         }
     }
 
+    public void addStudentIdToProjectName(Repository repo, ProgrammingExercise programmingExercise, Participation participation) {
+        // is Java programming language
+        if (programmingExercise.getProgrammingLanguage() == ProgrammingLanguage.JAVA) {
+            // Eclipse .project file
+            File eclipseProjectFile = new File(repo.getLocalPath().toString(), ".project");
```

Comment:
What happens, if the '.project' file is not in the root directory? Can we cover this case as well?

Reply:
This next [line](https://github.com/ls1intum/ArTEMiS/pull/491/files#diff-f4c3e69e07b0f66f35b2d3a3fce450eeR134) should cover the case

Reply:
I mean can we also support the rename task if the file is in a sub folder?

Reply:
done ✅